messaginginapp/src/main/java/io/customer/messaginginapp/type/InAppEventListener.kt

```diff
@@ -0,0 +1,8 @@
+package io.customer.messaginginapp.type
+
+interface InAppEventListener {
+    fun messageShown(message: InAppMessage)
+    fun messageDismissed(message: InAppMessage)
+    fun errorWithMessage(message: InAppMessage)
+    fun messageActionTaken(message: InAppMessage, currentRoute: String, action: String, name: String)
+}
```

Comment:
I think adding some docs will help. I'm not much familiar with gist, so not sure what to expect from `errorWithMessage` and `messageActionTaken` as a developer.

Reply:
What docs are you looking for? What questions do you have that docs can answer? 

The hard part is that I may not be the best person to write the docs, haha! I am also not that knowledgeable of Gist. I created this listener based on [the listener that already exists](https://gitlab.com/bourbonltd/gist-android/-/blob/467deb23be88fda089d0f3d85e23b3b1ba0f91fe/gist/src/main/java/build/gist/presentation/GistSdk.kt#L249-255).

Reply:
Lets maybe try to tackle this in the next PR?

Reply:
Yes, we can do this later.